core/eval.go

```diff
@@ -197,6 +197,29 @@ func evalSET(args []string) []byte {
 	return RESP_OK
 }
 
+// evalMSET puts multiple <key, value> pairs in db as in the args
+// args must contain key and value pairs.
+
+// Returns encoded error response if at least a <key, value> pair is not part of args
+// Returns encoded OK RESP once new entries are added
+// If the key already exists then the value will be overwritten and expiry will be discarded
+func evalMSET(args []string) []byte {
+	if len(args) <= 1 || len(args)%2 != 0 {
+		return Encode(errors.New("ERR wrong number of arguments for 'mset' command"), false)
+	}
+
+	// MSET does not have expiry support
+	var exDurationMs int64 = -1
+
+	for i := 0; i < len(args); i += 2 {
+		key, value := args[i], args[i+1]
+		oType, oEnc := deduceTypeEncoding(value)
```

Comment:
@soumya-codes do you think we should call `evalSET` instead of calling `Put` multiple times to the store directly? Using `evalSET` will ensure that any further changes to the `SET` command implementation would benefit this function as well.

Reply:
Unlike `SET` the `MSET` command has no optional arguments so invoking evalSET may not be appropriate here.
I am a unsure about the atomicity of the implementation though. 
We may have to handle this differently.
@JyotinderSingh wdyt?

Reply:
MSET needs to be atomic, let's introduce a store level API for this.

Reply:
> MSET needs to be atomic, let's introduce a store level API for this. Also, acquiring and releasing locks again and again may be too inefficient in general.

Cool. I was thinking about this only.